### PR TITLE
docs: apkam key naming conventions ADR

### PR DESCRIPTION
**Pull request by Xlin123**
**- What I did**
 create an ADR for apkam naming conventions
**- How I did it**
Filled out template.

Draft for now until we've been through review here.

**- How to verify it**

Rich diff

**- Description for the changelog**
feat: apkam key naming conventions ADR

### CODE REVIEW

**Comment by JeremyTubongbanua** [comment]
What are we hashing exactly for the `{hashed_namespace}` portion?

File: decisions/2024-07-apkam-key-conventions.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+* ### Remove the -k flag as a mandatory flag in at_activate enroll
+
+Come up with default key name for APKAM keys. The user will still be able
+to see which keys are for what using the `at_activate list` command
+
+Such as:
+* `@soccer0_{enrollment_id}_key.atKeys`
+* `@soccer0_{hashed_namespace}_key.atKeys`
+
+
```
Comment (xavierchanth) [inline decisions/2024-07-apkam-key-conventions.md]: Linter not happy with double newlines
(This was driving me nuts yesterday)

**Comment by xavierchanth** [comment]
> What are we hashing exactly for the `{hashed_namespace}` portion?

I had proposed this: for example, sshnp uses `sshnp:rw,sshrvd:rw`

So you can't just use the namespace (because there can be multiple, and with different permissions) but we could sort the namespaces alphanumerically and optionally hash the `sshnp:rw,sshrvd:rw` string to create a unique (enough) identifier for the keys.

The one thing I dislike about this current proposal is how to automatically detect apkam keys without having to specify -k with cli apps like npt, sshnp, sshnpd. Maybe if we use `sshnp-rw-sshrvd-rw` in the filename part, then the cli app could also look for that filename in addition to the default MPKAM file.

**Comment by Xlin123** [comment]
> Maybe if we use sshnp-rw-sshrvd-rw in the filename part, then the cli app could also look for that filename in addition to the default MPKAM file.

Thinking about worse case scenario, if these keys are compromised, the attacker would be able to see what apps they have access to. 

Regardless, I think this would be the best way to move forward, any other solution would include some extra layer of complexion that we have to include in source. This way, we can change less code and provide less complexity for our apps. 

Are we comfortable with including the namespace access in the file name?

File: decisions/2024-07-apkam-key-conventions.md (hunk 1 above)
Comment (Xlin123) [reply decisions/2024-07-apkam-key-conventions.md]: yea, odd that my local linter (literally the same one) doesn't autofix this...

**Review by xavierchanth — APPROVED** [review]
This can be merged if you would like and we can do edits + status update later

**Review by JeremyTubongbanua — APPROVED** [review]

**Pull request merged**